homeassistant/components/mqtt/subscription.py

```diff
@@ -58,6 +58,7 @@ async def subscribe(self) -> None:
         if not self.subscribe_task:
             return
         self.unsubscribe_callback = await self.subscribe_task
+        self.subscribe_task = None
```

Comment:
This looks unrelated to this PR?

Reply:
Right, we can take this one out, and make a fix for it. I ran into it when debugging. I don't think it is needed in this PR.

Reply:
Removed the unrelated change. I'll open a new PR for it if I think that it still makes sense.